relay/relaytest/mock_stats.go

```diff
@@ -160,11 +160,11 @@ func (m *MockStats) assertCallEqual(t testing.TB, expected *MockCallStats, actua
 		"Expectation must indicate whether RPC should succeed or fail.",
 	)
 
-	assert.Equal(t, expected.succeeded, actual.succeeded, "Unexpected number of successes.")
-	assert.Equal(t, expected.failedMsgs, actual.failedMsgs, "Unexpected reasons for RPC failure.")
-	assert.Equal(t, expected.ended, actual.ended, "Unexpected number of calls to End.")
+	failed := !assert.Equal(t, expected.succeeded, actual.succeeded, "Unexpected number of successes.")
+	failed = !assert.Equal(t, expected.failedMsgs, actual.failedMsgs, "Unexpected reasons for RPC failure.") || failed
+	failed = !assert.Equal(t, expected.ended, actual.ended, "Unexpected number of calls to End.") || failed
 
-	if t.Failed() {
+	if failed {
```

Comment:
This is so that the `if failed` condition is only triggered if _these_ assertions failed the test. Previously the condition would be satisfied if the test had previously failed for any reason, which was confusing when I was debugging, so I decided to change it to be based on these assertions specifically.

Does that make sense?